frontend/app/routes/protected/person-case/parent-details.tsx

```diff
@@ -0,0 +1,389 @@
+                          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:parent-details.province-error.invalid-province')),
+                        ),
+                      ),
+                      city: v.optional(
+                        v.pipe(
+                          v.string(t('protected:parent-details.city-error.required-city')),
+                          v.trim(),
+                          v.nonEmpty(t('protected:parent-details.city-error.required-city')),
+                          v.maxLength(maxStringLength, t('protected:parent-details.city-error.invalid-city')),
+                          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:parent-details.city-error.invalid-city')),
+                        ),
+                      ),
+                    }),
+                  ],
+                  t('protected:parent-details.country-error.required-country'),
+                ),
+              }),
+            ],
+            t('protected:parent-details.details-unavailable'),
+          ),
+          t('protected:parent-details.details-unavailable'),
+        ),
+        v.minLength(1),
+        v.maxLength(maxParents),
+      ) satisfies v.GenericSchema<ParentDetailsSessionData>;
+
+      const parentAmount = Number(formData.get('parent-amount')) || 0;
+      const inputLength = parentAmount > maxParents ? maxParents : parentAmount;
+
+      const input = Array.from({ length: inputLength }).map((_, i) => {
+        return {
+          unavailable: Boolean(formData.get(`${i}-unavailable`)),
+          givenName: String(formData.get(`${i}-given-name`)),
+          lastName: String(formData.get(`${i}-last-name`)),
+          birthLocation: {
+            country: String(formData.get(`${i}-country`)),
+            province: String(formData.get(`${i}-province`)),
+            city: String(formData.get(`${i}-city`)),
+          },
+        };
+      }) satisfies Partial<ParentDetailsSessionData>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.parentDetails = parseResult.output;
+
+      throw i18nRedirect('routes/protected/person-case/previous-sin.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
+}
+
+export default function CreateRequest({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const defaultFormValues = loaderData.defaultFormValues;
+  const [parentAmount, setParentAmount] = useState<number>(defaultFormValues?.length ?? 1);
```

Comment:
You don't have to explicitly type the `useState()` call when it can be inferred from the initial value.


```suggestion
  const [parentAmount, setParentAmount] = useState(defaultFormValues?.length ?? 1);
```